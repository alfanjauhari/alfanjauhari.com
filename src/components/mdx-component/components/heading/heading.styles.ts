import { styled } from '@/theme';

export const StyledH1 = styled('h1', {
  fontSize: '$xxxl',
  lineHeight: 1.13,
  fontWeight: 'bold',
  my: '$5',
});

export const StyledH2 = styled('h2', StyledH1, {
  fontSize: '$xxl',
  lineHeight: 1.1,
});

export const StyledH3 = styled('h3', StyledH1, {
  fontSize: '$xl',
  lineHeight: 1,
});

export const StyledH4 = styled('h4', StyledH3, {
  fontSize: '$lg',
});

export const StyledH5 = styled('h5', StyledH3, {
  fontSize: '$md',
});

export const StyledH6 = styled('h6', StyledH5, {
  fontSize: '$md',
  fontWeight: 'normal',
});
