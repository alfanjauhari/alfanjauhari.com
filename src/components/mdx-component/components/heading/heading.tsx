import { HTMLAttributes } from 'react';
import {
  StyledH1,
  StyledH2,
  StyledH3,
  StyledH4,
  StyledH5,
  StyledH6,
} from './heading.styles';

export type HeadingProps = HTMLAttributes<HTMLHeadingElement> & {
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
};

export function Heading({ as = 'h1', ...props }: HeadingProps) {
  const components = {
    h1: StyledH1,
    h2: StyledH2,
    h3: StyledH3,
    h4: StyledH4,
    h5: StyledH5,
    h6: StyledH6,
  };

  const HeadingComponent = components[as];

  return <HeadingComponent {...props} />;
}
