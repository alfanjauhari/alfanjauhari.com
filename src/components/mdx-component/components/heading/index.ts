export * from './heading';
export * from './heading.styles';
