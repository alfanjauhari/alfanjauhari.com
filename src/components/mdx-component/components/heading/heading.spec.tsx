import { render, screen } from '@testing-library/react';
import { Heading } from './heading';

describe('MDXComponent/Heading', () => {
  it('should render successfuly', () => {
    render(<Heading>Heading 1</Heading>);

    expect(
      screen.getByRole('heading', {
        name: 'Heading 1',
      }),
    ).toBeVisible();
  });

  it('should render correct html tag', () => {
    render(<Heading as="h2">Heading 2</Heading>);

    expect(
      screen.getByRole('heading', {
        level: 2,
      }),
    ).toBeVisible();
  });

  it('should match snapshot', () => {
    render(<Heading>Heading 1</Heading>);

    expect(screen.getByRole('heading')).toMatchSnapshot();
  });
});
