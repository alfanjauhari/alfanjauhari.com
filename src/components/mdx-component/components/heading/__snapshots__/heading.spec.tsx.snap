// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDXComponent/Heading should match snapshot 1`] = `
<h1
  class="c-gmKUXI"
>
  Heading 1
</h1>
`;
